src/FluxCalculator.jl

```diff
@@ -293,17 +260,26 @@ function compute_surface_fluxes!(
     boundary_space,
     thermo_params,
 )
+    # TODO lots of allocations here
     # `_int` refers to atmos state of center level 1
     z_int = Interfacer.get_field(atmos_sim, Val(:height_int))
     uₕ_int = Interfacer.get_field(atmos_sim, Val(:uv_int))
-    thermo_state_int = Interfacer.get_field(atmos_sim, Val(:thermo_state_int))
     z_sfc = Interfacer.get_field(atmos_sim, Val(:height_sfc))
 
+    # construct the atmospheric thermo states
+    thermo_state_atmos = TD.PhaseEquil_ρTq.(thermo_params, csf.ρ_atmos, csf.T_atmos, csf.q_atmos)
+
+    # construct the surface thermo state
+    # get surface air density by extrapolating atmospheric density to the surface
+    ρ_sfc = extrapolate_ρ_to_sfc.(Ref(thermo_params), thermo_state_atmos, T_sfc)
+
+    # compute surface humidity from the surface temperature, surface density, and phase
+    q_sfc = TD.q_vap_saturation_generic.(thermo_params, csf.T_sfc, ρ_sfc, get_field(sim, Val(:phase), thermo_params))
```

Comment:
use `FieldExchanger.compute_surface_humidity!` instead (move to FluxCalculator?)